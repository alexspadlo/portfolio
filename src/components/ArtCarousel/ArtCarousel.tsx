import React, { FC } from 'react'
import carData from './data'
import MyArtCarousel from './MyArtCarousel'
import GenButton from '@generic/GenButton'
import s from './ArtCarousel.scss'

const ArtCarousel: FC = () => {
    const { images } = carData || {}
    return (
        <div className='container-fluid my-5 py-4'>
            <div className='row py-5'>
                <div className='col-12 col-lg-7 m-auto'>
                    <section className='w-100 text-center'>
                        <p className={`${s.fontBold} h2`}>Artworks Showcase</p>
                        <p className={`${s.fontThin}`}>Art is my safe haven, where I flee in the sad days and also in the happy days to expose my inspirations.</p>
                        <GenButton content='View Artworks' onClick={() => console.log('artworks')} />
                    </section>
                </div>
            </div>
            <MyArtCarousel images={images} />
        </div>

    )
}

export default ArtCarousel