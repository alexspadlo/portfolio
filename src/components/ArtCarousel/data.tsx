// artworks
import artwork01 from '@images/artwork1.jpg'
import artwork02 from '@images/artwork2.jpg'
import artwork03 from '@images/artwork3.jpg'
import artwork04 from '@images/artwork4.jpg'
import artwork05 from '@images/artwork5.png'
import artwork06 from '@images/artwork6.png'
import artwork07 from '@images/artwork7.jpg'
import artwork08 from '@images/artwork8.jpg'
import artwork09 from '@images/artwork155.png'
import artwork010 from '@images/artwork9.png'
import artwork011 from '@images/artwork10.png'
import artwork012 from '@images/artwork11.png'
import artwork013 from '@images/artwork12.png'
import artwork014 from '@images/artwork13.png'
import artwork015 from '@images/artwork14.png'
import artwork016 from '@images/artwork15.png'

const images = [
    artwork01,
    artwork02,
    artwork03,
    artwork04,
    artwork05,
    artwork06,
    artwork07,
    artwork08,
    artwork09,
    artwork010,
    artwork011,
    artwork012,
    artwork013,
    artwork014,
    artwork015,
    artwork016,
]

const carData = {
    ref: 'artworks',
    images: images.map((cv, ind) => ({ index: ind, data: cv }))
}

export default carData