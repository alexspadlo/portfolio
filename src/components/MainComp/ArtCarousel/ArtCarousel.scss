@import "../../../stylesheet/_vars.scss";

.boxStyle {
  border: 1px solid #caa24f;
}

.centerAlign {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
}
