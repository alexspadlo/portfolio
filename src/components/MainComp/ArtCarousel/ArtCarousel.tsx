import React, { FC } from 'react'
import carData from './data'
import MyArtCarousel from './MyArtCarousel'
import GenButton from '@generic/GenButton'
import s from './ArtCarousel.scss'

interface ArtProps {
    handleDesign: (p?:string) => void
}

const ArtCarousel: FC<ArtProps> = ({ handleDesign }) => {
    const { images } = carData || {}
    return (
        <div className='container-fluid my-5 py-4 d-none d-lg-block'>
            <div className='row py-5'>
                <div className='col-12 col-lg-7 m-auto'>
                    <section className='w-100 text-center'>
                        <p className={`${s.fontBold} h2`}>Artworks Showcase</p>
                        <p className={`${s.fontThin}`}>Art is my safe haven, where I can escape from the pressures of daily life, focus on something positive and uplifting.</p>
                        <GenButton content='View All Artworks' onClick={() => handleDesign('artworks')} />
                    </section>
                </div>
            </div>
            <MyArtCarousel images={images} />
        </div>

    )
}

export default ArtCarousel