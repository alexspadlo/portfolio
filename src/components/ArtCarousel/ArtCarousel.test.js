import React from 'react'
import renderer from 'react-test-renderer'
import ArtCarousel from './ArtCarousel'

describe('Test on <ArtCarousel /> component', () => {
    test('Should render without the props', () => {
        const componentNoprops = renderer.create(<ArtCarousel />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
})
