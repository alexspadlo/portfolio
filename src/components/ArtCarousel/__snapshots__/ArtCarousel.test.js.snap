// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <ArtCarousel /> component Should render without the props 1`] = `
<div
  className="container-fluid my-5 py-4"
>
  <div
    className="row py-5"
  >
    <div
      className="col-12 col-lg-7 m-auto"
    >
      <section
        className="w-100 text-center"
      >
        <p
          className="fontBold h2"
        >
          Artworks Showcase
        </p>
        <p
          className="fontThin"
        >
          Art is my safe haven, where I flee in the sad days and also in the happy days to expose my inspirations.
        </p>
        <section
          className="w-100"
        >
          <span
            aria-hidden="true"
            className="d-block d-md-inline-block   textLink"
            onClick={[Function]}
            role="button"
          >
            View Artworks
          </span>
        </section>
      </section>
    </div>
  </div>
  <section
    className="w-100 position-relative"
  >
    <span
      className="topBorder"
    >
       
    </span>
    <div
      className="row"
    >
      <div
        className="col-sm text-center"
      >
        <section
          className="bg-white h-100 pointer"
          onClick={[Function]}
        >
          <img
            alt="test-file-stub"
            className="imgClass p-2"
            src="test-file-stub"
          />
        </section>
      </div>
      <div
        className="col-sm text-center"
      >
        <section
          className="bg-white h-100 pointer"
          onClick={[Function]}
        >
          <img
            alt="test-file-stub"
            className="imgClass p-2"
            src="test-file-stub"
          />
        </section>
      </div>
      <div
        className="col-sm text-center"
      >
        <section
          className="bg-white h-100 pointer"
          onClick={[Function]}
        >
          <img
            alt="test-file-stub"
            className="imgClass p-2"
            src="test-file-stub"
          />
        </section>
      </div>
      <div
        className="col-sm text-center"
      >
        <section
          className="bg-white h-100 pointer"
          onClick={[Function]}
        >
          <img
            alt="test-file-stub"
            className="imgClass p-2"
            src="test-file-stub"
          />
        </section>
      </div>
      <div
        className="col-sm text-center"
      >
        <section
          className="bg-white h-100 pointer"
          onClick={[Function]}
        >
          <img
            alt="test-file-stub"
            className="imgClass p-2"
            src="test-file-stub"
          />
        </section>
      </div>
    </div>
    <span
      className="bottomBorder"
    >
       
    </span>
  </section>
</div>
`;
